### DISCUSSION TITLE
Can UI components be prefilled from URL parameters?

### DISCUSSION THREAD

**Discussion by gcleaves** [windmill-labs/windmill#2764] [Questions]
Hi. Imagine I have a simple script that takes one string input. Is there any way to create a URL that points to the scripts so that input is prefilled when the user arrives on the page?

For example: http://localhost/scripts/get/8a16bb453d1ab37c?workspace=ws1&strInput=some_text

Thanks! Amazing platform!

Edit: For an App I used a frontend script with setTimeout to read the search parameters and setValue. This worked but feels brittle.

**Comment by rubenfiszel** [comment]
Note that query parameters are readable from ctx in apps. For scripts/flows page, it's a good idea, will add to backlog
